collective/dexteritytextindexer/converters.py

```diff
@@ -108,3 +109,15 @@ def convert(self):
         storage = self.field.interface(self.context)
         value = self.field.get(storage)
         return str(value)
+
+class TupleFieldConverter(DefaultDexterityTextIndexFieldConverter):
+    """Converts the data of a tuple field"""
+
+    implements(interfaces.IDexterityTextIndexFieldConverter)
+    adapts(IDexterityContent, ITuple, IWidget)
+
+    def convert(self):
+        """return the adapted field value"""
+        storage = self.field.interface(self.context)
+        value = self.field.get(storage)
+        return str(value)
```

Comment:
Since this handles tuple fields, the value is a tuple, right?
I think using the string representation of the tuple may not be optimal:
```python
>>> value = (u'Foo',)
>>> str(value)
"(u'Foo',)"
```

It would be better to join the values in the tuple (since we must return `str`  or [`unicode`](https://github.com/collective/collective.dexteritytextindexer/blob/244607b00de15064f043c3948778ac6b9714e531/collective/dexteritytextindexer/indexer.py#L103-L104)).
The problem is that we have no idea what value type the values of the tuple have; this can be set to any type with the `value_type` option.

I think it would be the safest to convert anything to `unicode`, but we must make sure that we have no encoding issues with bytestrings.
Example:

```python
storage = self.field.interface(self.context)
result = []
for value in self.field.get(storage):
    if isinstance(value, unicode):
        result.append(value)
    elif isinstance(value, str):
        result.append(value.decode('utf-8'))
    else:
        result.append(unicode(value))
return u' '.join(result)
```